### PR TITLE
Remove *.ts and *.js extension check code duplication

### PR DESCRIPTION
**Pull request by mjbvz**
**Defect**
There are currently a number of places that implement logic for checking if a file is Javascript or Typescript based on file extension. This code is hard to read and can get out of sync.

**Fix**
Created two helper methods to implement these checks. Replace most of the old extension check logic with calls to these helpers instead.

### CODE REVIEW

File: Nodejs/Product/Nodejs/NodejsFileExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+using System.IO;
+
+namespace Microsoft.NodejsTools {
+    static class NodejsFileExtensions {
```
Comment (mousetraps) [inline Nodejs/Product/Nodejs/NodejsFileExtensions.cs]: Can you change the name of this to be something like NodejsFileTypeHelpers? Files named with "Extensions" tend to be extension methods, so it's a bit overloaded.

**Comment by mousetraps** [comment]
There are also several more locations that are missed with this change that we may as well replace.

:+1: after that!

**Comment by mousetraps** [comment]
LGTM, but looks like there are conflicts now (might have been due to #730) - can you update and merge this in?

**Comment by mjbvz** [comment]
Closing since the line ending change makes merging back in to messy. Will reapply the change and submit another PR.

**Pull request closed without merging**